src/Stormpath/Resource/Application.php

```diff
@@ -281,6 +286,74 @@ public function authenticate($username, $password, array $options = array())
     }
 
 
+    public function createIdSiteUrl(array $options = array())
+    {
+        if( ! isset( $options['callbackUri'] ) )
+            throw new InvalidCallbackUriException('Please provide a \'callbackUri\' in the $options array.');
+
+        $client = Client::getInstance();
+
+        $p = parse_url ( $this->href );
+        $base = $p['scheme'] . '://' . $p['host'];
+
+        $apiId = $client->dataStore->getRequestExecutor()->getApiKey()->getId();
```

Comment:
I would probably move the ApiKey from the RequestExecutor to the DataStore and have a protected method to get the ApiKey from there. Just makes more sense to have it in the DataStore now.